src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java

```diff
@@ -0,0 +1,56 @@
+package com.exadel.discountwebapp.vendor.mapper;
+
+import com.exadel.discountwebapp.location.entity.Location;
+import com.exadel.discountwebapp.location.service.LocationService;
+import com.exadel.discountwebapp.vendor.entity.Vendor;
+import com.exadel.discountwebapp.vendor.vo.VendorRequestVO;
+import com.exadel.discountwebapp.vendor.vo.VendorResponseVO;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.modelmapper.convention.MatchingStrategies;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class VendorMapper {
+    private final LocationService locationService;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Autowired
+    public VendorMapper(LocationService locationService) {
+        this.locationService = locationService;
+        configureModelMapper();
+    }
+
+    public VendorResponseVO toVO(Vendor vendor) {
+        return modelMapper.map(vendor, VendorResponseVO.class);
+    }
+
+    public Vendor toEntity(VendorRequestVO request) {
+        return modelMapper.map(request, Vendor.class);
```

Comment:
should we set location here?

Reply:
yep, you are right. Thanks, that u have spotted this. "Setter" added.